### PR TITLE
sprint-6/issue-15#FireBase Fix 

### LINKED ISSUE #32: sprint-6/issue-15#FireBase Fix
## REQUIREMENTS / DESCRIPTION

> Fix new bug in FireBase

## DESIGN

> Null

### PR DESCRIPTION
**Pull request by FelixAva**
# Description

What did you implemented/Why did you implemented this:

- Fix HomeScreen issues
- Fix firebase connection issue
- Save user's data into firebase

## Testing

You can test it, Sign Up and  Sign In

### CODE REVIEW

File: .env
Comment (VISOSO2403) [inline .env:1]: Your dontenv is here, why?

Comment (FelixAva) [reply .env:1]: I forgot to add it to the .gitignore file. I'm changing it

File: src/components/CardCompanyComponent.tsx
Comment (VISOSO2403) [inline src/components/CardCompanyComponent.tsx:1]: change touchableOpacity, is imported from react- native dont from react-native-gesture-handler

File: src/components/CardCompanyComponent.tsx [hunk 1]
```diff
@@ -3,14 +3,15 @@ import {View, Image, StyleSheet} from 'react-native';
 import {TextComponent} from '../components';
 import {Company} from '../interface/companyinterface';
 import {globalStyles} from '../theme/globalTheme';
+import {TouchableOpacity} from 'react-native-gesture-handler';
```
Comment (VISOSO2403) [inline src/components/CardCompanyComponent.tsx]: change touchableOpacity, is imported from react- native dont from react-native-gesture-handler

File: src/components/DropDownComponent.tsx [hunk 2]
```diff
@@ -1,32 +1,35 @@
...
 
-const DropDownComponent = ({title}: {title: string}) => {
-  const [value, setValue] = useState('');
+const DropDownComponent = ({title, user}: {title: string; user: User}) => {
   const [isFocus, setIsFocus] = useState(false);
+  let genderValue = '';
 
   return (
     <View style={styles.container}>
       <Text style={styles.title}>{title}</Text>
```
Comment (VISOSO2403) [inline src/components/DropDownComponent.tsx]: use the TextComponet

File: src/hook/useAuth.tsx [hunk 3]
```diff
@@ -54,9 +58,14 @@ const useAuth = () => {
...
-    await auth()
-      .signOut()
-      .then(() => Alert.alert('Información', 'Se ha cerrado la sesión'));
+    try {
+      await GoogleSignin.signOut();
+      await auth()
+        .signOut()
+        .then(() => Alert.alert('Información', 'Se ha cerrado la sesión'));
+    } catch (error) {
+      console.error(error);
```
Comment (VISOSO2403) [inline src/hook/useAuth.tsx]: don't use console.log() use Alert, console.log is only for look errors on development

File: src/interface/db/UserInterface.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  photo: string;
+  working: boolean;
+
+  applications: Application[];
+  documents: Document[];
+}
+
+export interface Application {
+  company: string;
+  state: 'Pending' | 'Rejected' | 'Accepted';
```
Comment (VISOSO2403) [inline src/interface/db/UserInterface.ts]: don't use UPPERCASE on states

**Review by VISOSO2403 — CHANGES_REQUESTED** [review]
heed my comments and resolve conflicts


image

File: .env
Comment (VISOSO2403) [reply .env:1]: okey, don't worry😘

**Review by Antonio3veces3 — APPROVED** [review]
Well done guys!

**Review by VISOSO2403 — APPROVED** [review]
Nice job, only resolve conflicts

**Review by Jherrera11 — APPROVED** [review]
LGTM

**Comment by VISOSO2403** [comment]
I fixed conflicts, you need one more CR and aprove for merge brach

**Review by fpreciado3 — APPROVED** [review]
LGTM! 😁👍🙌🧀

**Pull request merged**